benchmark.h

```diff
@@ -322,12 +326,81 @@ class Benchmark {
                 << "," << searcher.name()
                 << std::endl;
     }
+    if (csv_) {
+      PrintResultCSV(index);
+    }
+  }
+
+  template<class Index>
+  void PrintResultCSV(const Index& index) {
+    const std::string filename = "./results/" + dataset_name_ + "_results_table.csv";
+
+    std::ofstream fout(filename, std::ofstream::out | std::ofstream::app);
+
+    if (!fout.is_open()) {
+      std::cerr << "Failure to print CSV on " << filename << std::endl;
+      return;
+    }
+
+    if (track_errors) {
+      fout << index.name()
+           << "," << index.variant()
+           << "," << log_sum_search_bound_
+           << "," << l1_sum_search_bound_
+           << "," << l2_sum_search_bound_
+           << std::endl;
+      return;
+    }
+
+    if (build) {
+      fout << index.name()
+           << "," << index.variant()
+           << "," << build_ns_
+           << "," << index.size()
+           << std::endl;
+      return;
+    }
+
+    if (cold_cache) {
+      double ns_per = ((double)individual_ns_sum) / ((double)lookups_.size());
+      fout << index.name()
+           << "," << index.variant()
+           << "," << ns_per
+           << "," << index.size()
+           << "," << build_ns_
+           << "," << searcher.name()
+           << std::endl;
+      return;
+    }
+
+    // print main results
+    std::ostringstream all_times;
+    for (unsigned int i = 0; i < runs_.size(); ++i) {
+      const double ns_per_lookup = static_cast<double>(runs_[i])
```

Comment:
It might be better to just output the median. Otherwise, we break the CSV format.

Reply:
Done